pandas/core/base.py

```diff
@@ -188,17 +188,32 @@ class SelectionMixin(object):
         builtins.max: 'max',
         builtins.min: 'min',
         np.sum: 'sum',
+        np.nansum: 'sum',
         np.mean: 'mean',
+        np.nanmean: 'mean',
         np.prod: 'prod',
         np.std: 'std',
+        np.nanstd: 'std',
         np.var: 'var',
+        np.nanvar: 'var',
         np.median: 'median',
+        np.nanmedian: 'median',
         np.max: 'max',
+        np.nanmax: 'max',
         np.min: 'min',
+        np.nanmin: 'min',
         np.cumprod: 'cumprod',
-        np.cumsum: 'cumsum'
+        np.nancumprod: 'cumprod',
+        np.cumsum: 'cumsum',
+        np.nancumsum: 'cumsum'
     }
 
+    # np.nanprod was added in np version 1.10.0, we currently support >= 1.9
+    try:
```

Comment:
If you have a preferred implementation for this, let me know and I'll happily change, explicitly checking version seemed ugly 😄

Reply:
you can just check _np_version_under1p10 and add it conditionally

Reply:
Awesome, didn't know this was a thing, thank you